Tasks 1 & 2: addition and subtraction BigUint

ewynx opened the pull request:
This implements the first 2 functions for BigUint, as well as defines the datastructure. (Note that addition uses @man2706kum's practice code as well!)

- BigUint definition
- Addition function
- Subtraction function
- Zero trait
- Multiple tests for both functions in main.sw

ewynx committed:
1: Addition BigUint

- Added datatype BigUint: vector of u32 limbs
- Implements addition for BigUint
- Tests for addition in main.sw (this is a script)

ewynx committed:
2: Subtraction BigUint

- Added subtraction for biguint
- With tests
- Introduced trait Zero, so we can check for "zero-being"
- Added tests for addition and structured them

ewynx committed:
Removed old TODO

man2706kum reviewed (changes requested):
Two comments have been added for changes. Please check it.

ewynx committed:
Fix PR comment: make sure the edge case of 2 elements of length 1 for subtraction is covered.
- i >= 0 instead of i > 0
- test added for 2 elements of len 1
- added code for edge case 0 - 0 = 0
- test added for 0 - 0 = 0
